src/Core/Framework/DataAbstractionLayer/Search/Criteria.php

```diff
@@ -645,6 +645,10 @@ private function collectFields(array $parts): array
      */
     private function validateIds(array $ids): void
     {
+        if (count($ids) === 0) {
```

Comment:
I would consider this a break and this needs to be surrounded with the major feature flag. what do you think @AydinHassan ?

Reply:
@mitelg the team agreed that the check was missed and therefore was a mistake. From our perspective we would like to proceed as a bug fix, is it okay with you?

Reply:
is okay, I had another look today and it seems the check is just moved. everything fine :+1: